packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -641,6 +641,19 @@ export function getShadowRootVM(root: ShadowRoot): VM {
     return getHiddenField(root, ViewModelReflection) as VM;
 }
 
+function hasDifferentIdentities(a: VNode | null, b: VNode | null): boolean {
+    if (a === b) {
```

Comment:
this could be collided into one single return statement, but I feel that we should reasoning about it first.